R/swagger.R

```diff
@@ -191,5 +346,63 @@ removeNaOrNulls <- function(x) {
   ret <- lapply(x, removeNaOrNulls)
   class(ret) <- class(x)
 
+  # Put example back in
+  if (saveExample) {
+    ret$example <- savedExample
+  }
+
   ret
 }
+
+#' For openapi definition
+#' @noRd
+priorizeProperty <- function(...) {
+  l <- list(...)
+  if (length(l) > 0L) {
+    isnullordefault <- vapply(l, function(x) {isNaOrNull(x) || isTRUE(attributes(x)$default)}, logical(1))
+    # return the position of the first FALSE value or position 1 if all values are TRUE
+    return(l[[which.min(isnullordefault)]])
+  }
+  return()
+}
+
+#' Check if x is JSON serializable
+#' @noRd
+isJSONserializable <- function(x) {
+  testJSONserializable <- TRUE
+  tryCatch(toJSON(x),
+           error = function(cond) {
+             # Do we need to test for specific errors?
+             testJSONserializable <<- FALSE}
+  )
+  testJSONserializable
+}
+
+#' Extract metadata on args of plumberExpression
+#' @noRd
+getArgsMetadata <- function(plumberExpression){
+  #return same format as getTypedParams or params?
+  args <- formals(eval(plumberExpression))
+  required_arg <- formals(function(x){})$x
+  lapply(args[!names(args) %in% c("...", "res", "req")], function(arg) {
+    required <- identical(arg, required_arg)
+    if (is.call(arg) || is.name(arg)) {
+      arg <- tryCatch(
+        eval(arg),
```

Comment:
This likely needs to be evaluate in the environment of the function

Reply:
Hopefully not as it is supposed to capture arguments value at the call level before the function is executed.

Say:
```r
avalue <- c(1,2)
function(a = avalue) {
 something
}
```

It should eval `a` to `c(1,2)`

Reply:
added envir